### DISCUSSION TITLE
Creating union types from prismaObjects

### DISCUSSION THREAD

**Discussion by DerTimonius** [hayes/pothos#934] [Q&A]
I'm working on a project that uses pothos in conjunction with Prisma. Currently, I need to implement a mutation that can return varying types depending on the input - in my case `AnswerText` or `AnswerDate` which are both `prismaObjects`. 
Following the [guide on unions](https://pothos-graphql.dev/docs/guide/unions) I tried to implement a `AnswerUnion` type like this:
```ts
const AnswerUnion = builder.unionType('AnswerUnion', {
  types: [AnswerDate, AnswerText],
});
```
This is not showing any TypeScript errors at this level, but when I'm trying to use the type I'm getting errors stating that properties like `modelName` are missing, even if I add a `resolveTypes` function to the unionType.

Is it possible to create `unionTypes` out of `prismaObjects` or is it obligatory to use `objectTypes` in this case?

**Comment by hayes** [comment]
It's possible to create union types with prisma objects, but the query optimization won't work quite as well through unions.  There is an undocumented trick you need to use with unions (documenting this better has been on my todo list for a while):

```ts
answers t.field({
  type: [AnswerUnion],
  resolve: async () => {
    const dates = getDateAnswers();
    const texts = getTextAnswers();
    return [...AnswerDate.addBrand(dates), ...AnswerText.addBrand(texts)];
  },
}),
```
